.github/workflows/state-compatibility-check.yml

```diff
@@ -0,0 +1,193 @@
+# This workflow checks that a specific commit / branch / tag is state compatible 
+# with the latest osmosis version by:
+
+# - building the new `osmosisd` binary with the latest changes
+# - replaying a configurable number of previous blocks from chain history
+
+# Currently, the node starts from a snapshot taken some blocks before the last epoch 
+# and waits `DELTA_HALT_HEIGHT` blocks after epoch before finally halting.
+
+# Important Caveat:
+
+# The fact that this workflow succeeds and the binary doesn't fail doesn't 
+# directly imply that the new binary is state-compatible.
+# It could be that the binary is not state-compatible, but the condition 
+# which would break state compatibility was not present in the chunk of block history used.
+
+# On the other hand, if the workflow fails, the binary is not state-compatible.
+
+name: Check state compatibility
+
+# ************************************ NOTE ************************************
+# 
+# DO NOT TRIGGER THIS WORKFLOW ON PUBLIC FORKS
+#
+# This workflow runs on a self-hosted runner and forks to this repository 
+# can potentially run dangerous code on the self-hosted runner machine 
+# by creating a pull request that executes the code in a workflow.
+# 
+# ******************************************************************************
+
+on:
+  pull_request:
+    branches:
+      - 'v[0-9]+.x'
+  schedule:
+      - cron: '01 23 * * *' # Runs at 01:23 UTC every day
+
+env:
+  GENESIS_URL: https://github.com/osmosis-labs/networks/raw/main/osmosis-1/genesis.json
+  SNAPSHOT_BUCKET: https://osmosis-snapshot.sfo3.cdn.digitaloceanspaces.com
+  RPC_ENDPOINT: https://rpc.osmosis.zone
+  LCD_ENDPOINT: https://lcd.osmosis.zone
+  DELTA_HALT_HEIGHT: 25
+ 
+jobs:
+
+  check_state_compatibility:
+    # DO NOT CHANGE THIS: please read the note above
+    if: ${{ github.event_name == 'schedule' || github.event.pull_request.head.repo.full_name == 'osmosis-labs/osmosis' }}
+    runs-on: osmo-runner 
+    steps:
+      - 
+        # If workflow was triggered by a schedule, 
+        # explicitly checkout the latest branch
```

Comment:
```suggestion
        # If workflow was triggered by a schedule, explicitly checkout the latest branch
        # otherwise checkout the release branch were targetting
```